custom/icds_reports/reports/ucr.py

```diff
@@ -339,5 +339,96 @@ def get_total_records(self):
             return session.connection().execute(count_query, **self.helper.sql_alchemy_filter_values).scalar()
 
 
+CcsRecordMonthlyViewAlchemy = sqlalchemy.Table(
+    'ccs_record_monthly_view', metadata, autoload=True
+)
+
+
+class CcsRecordMonthlyUCR(ConfigurableReportCustomQueryProvider):
+    def __init__(self, report_data_source):
+        self.report_data_source = report_data_source
+        self.helper = self.report_data_source.helper
+        self.helper.set_table(CcsRecordMonthlyViewAlchemy)
+
+    @property
+    def table(self):
+        return CcsRecordMonthlyViewAlchemy
+
+    def _get_query_object(self, total_row=False):
+
+        def format_column(title, logic):
+            return func.count(self.table.c.case_id).filter(logic).label(title)
+
+        pregnant = self.table.c.pregnant == 1
+        lactating = self.table.c.lactating == 1
+        st_caste = self.table.c.caste == 'st'
+        sc_caste = self.table.c.caste == 'sc'
+        rations_gte_21 = self.table.c.num_rations_distributed >= 21
+        rations_lt_21 = self.table.c.num_rations_distributed < 21
```

Comment:
I think that this should also be > 0. Otherwise this will double report the absent women

Reply:
https://github.com/dimagi/commcare-hq/pull/23552/commits/8ba02a263fbefa4bd6af7a4b71c09dc68dd8aaae